domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+            corresponds to clockwise (CW) and the Normally Closed (NC)
+            position corresponds to counterclockwise (CCW).
+        DIRECTION_RELAY_PIN_NUMBER : int
+            The GPIO pin that corresponds to the direction relay on the
+            automationHAT (relay 2).
+        BOUNCE_TIME : float
+            A buffer period (in seconds) where home/encoder input will ignore
+            additional (de)activation.
+
+        """
+
+        if testing:
+            # Set the default pin factory to a mock factory
+            Device.pin_factory = MockFactory()
+            # in case a previous instance has been initialised, tell the
+            # pin factory to release all the pins
+            Device.pin_factory.reset()
+
+            # set a timeout length in seconds for wait_for_active() calls
+            WAIT_TIMEOUT = 1
+
+            # in testing mode we need to create a seperate pin object so we can
+            # simulate the activation of our fake DIDs and DODs
+            self.encoder_pin = Device.pin_factory.pin(ENCODER_PIN_NUMBER)
+            self.home_sensor_pin = Device.pin_factory.pin(
+                HOME_SENSOR_PIN_NUMBER)
+        else:
+            # set the timeout length variable to None for non testing mode
+            WAIT_TIMEOUT = None
+
+        # set a wait time for testing mode that exceeds BOUNCE_TIME
+        self.test_mode_delay_duration = BOUNCE_TIME + 0.05
+        # set the timeout for wait_for_active()
+        self.wait_timeout = WAIT_TIMEOUT
+
+        # led_status is set with binary number, each zero/position sets the
+        # state of an LED, where 0 is off and 1 is on
+        self.led_status = 0b000000000000000000
+        # create a dictionary of the LED name as the key and the digit of
+        # the self.led_status binary integer it corresponds to. This means
+        # we can convert the binary integer to a string and use the index
+        # to change a 0 to 1 and vice versa and then convert back to a
+        # binary integer. The updated self.led_status can then be sent to
+        # the LED controller.
+        self.led_lights_ind = {
+            'power': 2,
+            'comms': 3,
+            'warn': 4,
+            'input_1': 5,
+            'input_2': 6,
+            'input_3': 7,
+            'relay_3_normally_closed': 8,
+            'relay_3_normally_open': 9,
+            'relay_2_normally_closed': 10,
+            'relay_2_normally_open': 11,
+            'relay_1_normally_closed': 12,
+            'relay_1_normally_open': 13,
+            'output_3': 14,
+            'output_2': 15,
+            'output_1': 16,
+            'adc_3': 17,
+            'adc_2': 18,
+            'adc_1': 19
+        }
+        if debug_lights:  # pragma: no cover
+            # if we are actually using the debug lights we can enable them now
+            sn3218.output([0x10] * 18)
+            sn3218.enable_leds(self.led_status)
```

Comment:
yea, I think what I was originally doing was turning off the lights for when we run with real hardware but dont want the lights, as they get turned on when you import the sn3218 led controller library. Should just import that library from within a `if debug_lights:` clause instead of doing it with the other module exports.